spec/models/sorn_spec.rb

```diff
@@ -78,6 +79,48 @@
     end
   end
 
+  describe ".update_mentioned_sorns" do
+    let(:sorn) { create :sorn, xml: file_fixture("sorn.xml").read }
+    let!(:child_sorn) { Sorn.create(citation: "01 FR 1234") }
+
+    before { sorn.update_mentioned_sorns }
+
+    it "finds FR citations in the xml that are SORNs" do
+      expect(sorn.mentioned).to eq [child_sorn]
+    end
+
+    it "also adds the parent id to the child mentions" do
+      expect(child_sorn.mentioned).to eq [sorn]
+    end
+
+    context "with existing mentioned array" do
+      let(:existing_child_sorn) { create :sorn, citation: "something else" }
+      let(:sorn) { create :sorn, mentioned: [existing_child_sorn], xml: file_fixture("sorn.xml").read }
+
+      it "adds to the array" do
+        expect(sorn.mentioned).to eq [existing_child_sorn, child_sorn]
+      end
+
+      it "doesn't duplicate mentions though" do
+        sorn.update_mentioned_sorns
+        sorn.update_mentioned_sorns
+        sorn.update_mentioned_sorns
+
+        expect(sorn.mentioned).to eq [existing_child_sorn, child_sorn]
+      end
+    end
```

Comment:
seems like it would be good to have a test case of a citation that isn't a sorn and to expect that its not in the mentioned_sorns

Reply:
I added two citations to the sorn.xml fixture. https://github.com/18F/all_sorns/pull/43/files#diff-427ff58ad37a4b713b8ccb528882c48dd3d008ad0f9f41068b8851e055cbcea0R135

Reply:
ah ok. looks good then.